client/components/photo/dropZone.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import Dropzone from 'react-dropzone'
+import FlatButton from 'material-ui/lib/flat-button';
+
+const DropZone = ({
```

Comment:
I think the reason DropZone wasn't re-rendering on upload is bc it's re-render method isn't subbed to the store. You would either need to hook it up with connect, or subscribe/unsubscribe in life cycle methods